internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -269,7 +269,7 @@ func (r *Reconciler) reconcileMachineHealthCheck(ctx context.Context, current, d
 
 		log.Infof("Creating %s", tlog.KObj{Obj: desired})
 		if err := r.Client.Create(ctx, desired); err != nil {
-			return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: desired})
+			return createError(err, desired)
```

Comment:
for sake of consistency, should this be
```suggestion
			errors.Wrapf(err, "failed to create %s", desired.Kind)
```
same below/in other places where we are returning createError without wrapping

Reply:
I think we have to decide if we want to add the "failed to create" here or in the func.

I think it could be better to do it here (as we can provide additional context which is needed in some cases). But then the createError/apiErrorWithoutObjectName should focus on sanitizing the StatusError and not also wrap it with `failed to create ...`.

Reply:
For now I'd prefer to keep the create specific information here - I don't think this is more than a single-use function. The additional info can be added by wrapping errors I think - e.g. where the MachineDeployment object kind is added.

Reply:
Sure, just give it a try.